libraries/core_libs/node/src/node.cpp

```diff
@@ -232,24 +232,25 @@ void FullNode::start() {
   if (conf_.network.network_is_boot_node) {
     LOG(log_nf_) << "Starting a boot node ..." << std::endl;
   }
-  if (!conf_.test_params.rebuild_db) {
-    network_->start();
-    blk_proposer_->setNetwork(network_);
-    blk_proposer_->start();
-  }
+
   vote_mgr_->setNetwork(network_);
   pbft_mgr_->setNetwork(network_);
   dag_mgr_->setNetwork(network_);
-  pbft_mgr_->start();
-  dag_mgr_->start();
 
   if (conf_.test_params.rebuild_db) {
     rebuildDb();
     LOG(log_si_) << "Rebuild db completed successfully. Restart node without db_rebuild option";
     started_ = false;
     return;
+  } else {
+    network_->start();
+    blk_proposer_->setNetwork(network_);
+    blk_proposer_->start();
   }
 
+  pbft_mgr_->start();
+  dag_mgr_->start();
```

Comment:
are you sure this is not needed during rebuild? did you try to rebuild DB ?

Reply:
There is a race condition when pbft start running during rebuild DB.
If only one node initial with delegation when deploy network. And the node send out coins to other nodes later (like many unit tests cases). When the node rebuild DB, if start PBFT running that will cause finalize new pbft blocks very quick since we introduce pbft produce empty blocks. That will cause forking with old PBFT chain in old DB, and failed rebuild DB. I fixed it in this commit https://github.com/Taraxa-project/taraxa-node/pull/1612/commits/7e3c468b7955f076ecc0f92004e63eac4097bf0c

And we have rebuild DB unit test here https://github.com/Taraxa-project/taraxa-node/blob/develop/tests/full_node_test.cpp#L1325